src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelEMParams.java

```diff
@@ -0,0 +1,1179 @@
+            shortName = PSI_UPDATE_ENABLED_SHORT_NAME,
+            fullName = PSI_UPDATE_ENABLED_LONG_NAME,
+            optional = true
+    )
+    protected boolean psiUpdateEnabled = DEFAULT_PSI_UPDATE_ENABLED;
+
+    @Advanced
+    @Argument(
+            doc = "Enable adaptive switiching of Psi solver modes (isotropic to target-resolved)",
+            shortName = ADAPTIVE_PSI_SOLVER_MODE_SWITCHING_ENABLED_SHORT_NAME,
+            fullName = ADAPTIVE_PSI_SOLVER_MODE_SWITCHING_ENABLED_LONG_NAME,
+            optional = true
+    )
+    protected boolean adaptivePsiSolverModeSwitchingEnabled = DEFAULT_ADAPTIVE_PSI_SOLVER_MODE_SWITCHING_ENABLED;
+
+    @Argument(
+            doc = "Enable copy ratio updates",
+            shortName = CR_UPDATE_ENABLED_SHORT_NAME,
+            fullName = CR_UPDATE_ENABLED_LONG_NAME,
+            optional = true
+    )
+    protected boolean copyRatioUpdateEnabled = DEFAULT_CR_UPDATE_ENABLED;
+
+    @Argument(
+            doc = "RDD checkpointing path (required if RDD checkpointing is enabled)",
+            shortName = RDD_CHECKPOINTING_PATH_SHORT_NAME,
+            fullName = RDD_CHECKPOINTING_PATH_LONG_NAME,
+            optional = true
+    )
+    protected String rddCheckpointingPath = DEFAULT_RDD_CHECKPOINTING_PATH;
+
+    @Argument(
+            doc = "Model checkpointing path (required if model checkpointing is enabled)",
+            shortName = RUN_CHECKPOINTING_PATH_SHORT_NAME,
+            fullName = RUN_CHECKPOINTING_PATH_LONG_NAME,
+            optional = true
+    )
+    protected String runCheckpointingPath = DEFAULT_RUN_CHECKPOINTING_PATH;
+
+    @Advanced
+    @Argument(
+            doc = "Enable extended posterior output",
+            shortName = EXTENDED_POSTERIOR_OUTPUT_ENABLED_SHORT_NAME,
+            fullName = EXTENDED_POSTERIOR_OUTPUT_ENABLED_LONG_NAME,
+            optional = true
+    )
+    protected boolean extendedPosteriorOutputEnabled = DEFAULT_EXTENDED_POSTERIOR_OUTPUT_ENABLED;
+
+    @Advanced
+    @Argument(
+            doc = "Number of target space paritions (for spark mode)",
+            shortName = NUMBER_OF_TARGET_SPACE_PARTITIONS_SHORT_NAME,
+            fullName = NUMBER_OF_TARGET_SPACE_PARTITIONS_LONG_NAME,
+            optional = true
+    )
+    protected int numTargetSpaceParititions = DEFAULT_NUMBER_OF_TARGET_SPACE_PARTITIONS;
+
+    /**********************************************************************************
+     * setters and getters (used for on-the-fly update of parameters and for testing) *
+     **********************************************************************************/
+
+    public CoverageModelEMParams setMaxEMIterations(final int maxEMIterations) {
+        this.maxEMIterations = ParamUtils.isPositive(maxEMIterations, "Maximum EM iterations must be positive.");
+        return this;
+    }
+
+    public int getMaxEMIterations() { return maxEMIterations; }
+
+    public CoverageModelEMParams setNumLatents(final int numLatents) {
```

Comment:
For example, this is one of the few setters that is actually used (to check whether the requested number of latent variables matches that used in a previous run).  In that case, I would simply perform the check, log a message stating how many latent variables will actually be used and why, and then store the actual number used in a field of the object that is produced by the run.  So this argument collection should not be responsible for holding mutable state and should be treated as essentially immutable input from the CLI.